src/gluonts/model/predictor.py

```diff
@@ -241,11 +247,12 @@ def __init__(
         freq: str,
         ctx: mx.Context,
         input_transform: Transformation,
+        lead_time: int = 0,
         forecast_generator: ForecastGenerator = SampleForecastGenerator(),
         output_transform: Optional[OutputTransform] = None,
         dtype: DType = np.float32,
     ) -> None:
-        super().__init__(prediction_length, freq)
+        super().__init__(prediction_length, freq, lead_time)
```

Comment:
Looking at this, I don't know if I like the order. We could force named arguments to make things more explicit.

There could be other ways, for example we have an object defining the forecast horizon:

```python
class ForecastHorizon:
    prediction_length: int
    lead_time: int = 0
```

Overall, I think I would like to be able to change the API if we found a way which we like.

What if an algorithm can predict any range of future values; i.e. why do these values need to be fixed? Also, why do we have to pass freq?

Reply:
> What if an algorithm can predict any range of future values; i.e. why do these values need to be fixed? Also, why do we have to pass freq?

Sure, one could think of having models that predict, say, 10, 12, and 19 time steps from now. For the moment, this proposal makes the way the prediction range is specified more general than the status quo by allowing shifting the prediction range ahead. To me, this is already very general, but in the future we can decide to change the way this information is encoded, and potentially make it even more general.

Reply:
I think I would opt for using named_arguments now, and be ready to make some breaking changes later.

Reply:
agreed, named argument it is